staging/src/k8s.io/client-go/tools/portforward/portforward_test.go

```diff
@@ -83,6 +82,17 @@ func TestParsePortsAndNew(t *testing.T) {
 				{protocol: "tcp6", address: "::1", failureMode: "all"},
```

Comment:
I wonder if it's a bug or if we don't care. 

For this testcase, when address `"localhost"` was processed, the "output" addresses should have been
```
{protocol: "tcp4", address: "127.0.0.1", failureMode: "all"},
{protocol: "tcp6", address: "::1", failureMode: "all"},
```
However the consequent address `"127.0.0.1"` gets processed and overrides the `failureMode` of 127.0.0.1 to `"any"`.

I can imagine the reverse case can also happen. It's orthogonal to this PR so I won't hold it

Reply:
@rowleyajI this is related to [this conversation](https://github.com/kubernetes/kubernetes/pull/46517/files#r179165644) in the original PR about how to handle failures and it seems to be expected behaviour:

* if someone specifies the addresses explicitly - then command should fail if any of the address fail (covered by this test case)
* If no addresses specified explicitly (`localhost`), then allow to silently fail one of them

Reply:
Right. I'm not convinced that the current implementation is the final goal based on the conversation. But for the current implementation, imagine the two scenarios: 

1. input addresses: `"localhost", "127.0.0.1", "::1"`; output addresses would be 
```
{protocol: "tcp4", address: "127.0.0.1", failureMode: "any"},
{protocol: "tcp6", address: "::1", failureMode: "any"},
```
2. input addresses: `"127.0.0.1", "::1", "localhost"`; output addresses would be 
```
{protocol: "tcp4", address: "127.0.0.1", failureMode: "all"},
{protocol: "tcp6", address: "::1", failureMode: "all"},
```

I'm thinking open an issue to continue the discussion in https://github.com/kubernetes/kubernetes/pull/46517/files#r179165644

Reply:
Yeah, you are right. The special case for `localhost` at the end overrides previously set values.

Please, cc me on the issue, once created.